heartbeat/VirtualDomain

```diff
@@ -608,28 +656,18 @@ VirtualDomain_Migrate_To() {
 		if [ -n "${OCF_RESKEY_migration_transport}" ]; then
 			transport_suffix="+${OCF_RESKEY_migration_transport}"
 		fi
-		# A typical migration URI via a special  migration network looks
-		# like "tcp://bar-mig:49152". The port would be randomly chosen
-		# by libvirt from the range 49152-49215 if omitted, at least since
-		# version 0.7.4 ...
-		if [ -n "${OCF_RESKEY_migration_network_suffix}" ]; then
-			hypervisor="${OCF_RESKEY_hypervisor%%[+:]*}"
-			# Hostname might be a FQDN
-			migrate_target=$(echo ${target_node} | sed -e "s,^\([^.]\+\),\1${OCF_RESKEY_migration_network_suffix},")
-			case $hypervisor in
-				qemu)
-					# For quiet ancient libvirt versions a migration port is needed
-					# and the URI must not contain the "//". Newer versions can handle
-					# the "bad" URI.
-					migrateuri="tcp:${migrate_target}:${OCF_RESKEY_migrateport}"
-					;;
-				xen)
-					migrateuri="xenmigr://${migrate_target}"
-					;;
-				*)
-					ocf_log warn "$DOMAIN_NAME: Migration via dedicated network currently not supported for ${hypervisor}."
-					;;
-			esac
+
+		# User defined migrateuri or do we make one?
+		migrate_opts="$OCF_RESKEY_migrate_options"
+		if echo "$migrate_opts" | fgrep -qs -- "--migrateuri="; then
```

Comment:
On Wed, Jan 28, 2015 at 09:19:56AM -0800, David Vossel wrote:
> > -					# For quiet ancient libvirt versions a migration port is needed
> > -					# and the URI must not contain the "//". Newer versions can handle
> > -					# the "bad" URI.
> > -					migrateuri="tcp:${migrate_target}:${OCF_RESKEY_migrateport}"
> > -					;;
> > -				xen)
> > -					migrateuri="xenmigr://${migrate_target}"
> > -					;;
> > -				*)
> > -					ocf_log warn "$DOMAIN_NAME: Migration via dedicated network currently not supported for ${hypervisor}."
> > -					;;
> > -			esac
> > +
> > +		# User defined migrateuri or do we make one?
> > +		migrate_opts="$OCF_RESKEY_migrate_options"
> > +		if echo "$migrate_opts" | fgrep -qs -- "--migrateuri="; then
> 
> when did --migrateuri become an option. In older versions of virsh it looks like migrate uri is expected to proceed the destination uri in the virsh command.

I could find it v1.0.5. Otherwise, I don't know when it appeared.
Anyway, I'll take it out as proposed in comments.